kinto/views/__init__.py

```diff
@@ -29,4 +30,10 @@ def object_exists_or_404(request, collection_id, object_id, parent_id=''):
                            object_id=object_id)
     except exceptions.RecordNotFoundError:
         # XXX: We gave up putting details about parent id here (See #53).
-        raise httpexceptions.HTTPNotFound()
+        detail_dict = {
+                "id": collection_id,
+                "resource_name": "collection"
+            }
```

Comment:
The resource name is one of `bucket`, `collection`, `group`, `record`. Here by using `collection_id` it would work as expected. The name of this parameter is very bad I know, because it seems related to collections only, but it actually represent the resource id/name. In #710 we would fix that naming problem.